crates/autopilot/src/domain/competition/participation_guard/db.rs

```diff
@@ -0,0 +1,123 @@
+use {
+    crate::{
+        database::Postgres,
+        domain::{eth, Metrics},
+        infra,
+    },
+    ethrpc::block_stream::CurrentBlockWatcher,
+    std::{
+        collections::HashMap,
+        sync::Arc,
+        time::{Duration, Instant},
+    },
+};
+
+/// Checks the DB by searching for solvers that won N last consecutive auctions
+/// but never settled any of them.
+#[derive(Clone)]
+pub(super) struct Validator(Arc<Inner>);
+
+struct Inner {
+    db: Postgres,
```

Comment:
nit: can we substitute this with

`persistence: infra::Persistence`

and move related db function to this infra module? This is how we decided to do things for all components inside `autopilot::domain`